src/main/java/io/openapitools/swagger/GenerateMojo.java

```diff
@@ -77,6 +80,14 @@
     @Parameter(defaultValue = "false")
     private boolean attachSwaggerArtifact;
 
+    /**
+     * Specifies the implementation of {@link Application}. If the class is not specified,
+     * the resource packages are scanned for the {@link Application} implementations
+     * automatically.
+     */
+    @Parameter(name = "applicationClass", defaultValue = "")
+    private String applicationClass;
```

Comment:
I am a bit unsure as to whether Maven would actually do the classloading if the type was speciffied as `Class<...>` instead of `String`...

Reply:
I have no idea about this one. Also, a question is which classloader would Maven use to do the classloading.

Reply:
Good point - lets leave it as-is